packages/core/src/chain.ts

```diff
@@ -74,14 +75,19 @@ async function setChain(options: {
       code === ProviderRpcErrorCode.UNRECOGNIZED_CHAIN_ID
     ) {
       // chain has not been added to wallet
+      if (rpcUrl || label || token) {
+        // If values passed, update chain in state
+        chain.rpcUrl = rpcUrl
+        chain.label = label
+        chain.token = token
+        updateChain(chain)
+      }
```

Comment:
This could be problematic if say the `rpcUrl` was set on init, but then not included in the call to`setChain`, it would get overwritten with `undefined`.
We might to just need to be a bit verbose and check if each value exists before assigning it.

```suggestion
      if (rpcUrl) {
        chain.rpcUrl = rpcUrl
      }

      if (label) {
        chain.label = label
      }
      
      if (token) {
        chain.token = token
      }
            
      updateChain(chain)
```

Another option would be to make `chain` a `let` var so we can reassign it and collect the additional chain update vars using the rest operator on line 27:

```javascript
const { chainId, chainNamespace = 'evm', wallet: walletToSet, ...chainUpdates } = options
```

and then merge them with `chain`

```javascript
if (chainUpdates) {
  chain = { ...chain, ...chainUpdates }
  updateChain(chain)
}
```

If we went this route, we'd just need to be careful if we add parameters to the `setChain` options as they would get merged in as well.

Reply:
ah good catch! Thanks for the suggestion!